index.html

```diff
@@ -138,233 +137,254 @@
-
-          if (code == 13) {
-            e.preventDefault();
-
-            if ($(this).val()) {
-              loadPaste($(this).val());
-            } else {
-              alert("No ID!");
-            }
-          }
-        });
-
-        $("#uuid").keypress(function(e) {
-          var code = e.keyCode || e.which;
-
-          if (code == 13) {
-            e.preventDefault();
-
-            if ($(this).val()) {
-              updatePaste($(this).val());
-            } else {
-              alert("No UUID!");
-            }
-          }
-        });
-
-        $("#expire").on("input", function(e) {
-          var v = $(this).val();
-
-          if (v == 1) {
-            $("#hours").text("hour");
-          } else if (v == 0) {
-            $(this).val("");
-            $("#hours").text("hours");
-          } else {
-            $("#hours").text("hours");
-          }
-        });
-      });
+    <div id="paste_error" hidden="true">
+      <ul id="error_list"></ul>
+    </div>
+
+    <script>
+     $(document).ready(function() {
+       document.title = "pb";
+
+       $("a").attr("title", "Opens in a new tab");
+
+       $("button").addClass("btn btn-default");
+
+       // button colors
+       $("button#delete").addClass("btn-danger");
+       $("button#submit").addClass("btn-primary");
+       $("button#clear").addClass("btn-danger");
+
+
+       $("#file").hide();
+
+       $("#clear").click(function(e) {
+         e.preventDefault();
+
+         $("#text").val("");
+         $("#private").prop("checked", false);
+         $("#expire").val("");
+         $("#file").val("");
+         $("#filename").text("");
+         $("#pasteid").val("");
+         $("#uuid").val("");
+
+         $("#paste_response").hide();
+         $("#text").focus();
+       });
+
+       function getData() {
+         data = {
+           c: $("#text").val()
+         }
+
+         if ($("#private").prop("checked")) {
+           data.p = 1;
+         }
+
+         if ($("#expire").val() > 0) {
+           data.s = $("#expire").val() * 3600;
+         }
+
+         return data;
+       }
+
+       function displayResponse(response) {
+         $("#paste_error").hide();
+         $("#paste_response").show();
+
+         $("#response_url").attr("href", response.url);
+         $("#response_url").text(response.url);
+         $("#response_status").text(response.status);
+         $("#response_uuid").text(response.uuid);
+         $("#pasteid").val(response.short);
+
+         if (response.uuid) {
+           $("#uuid").val(response.uuid);
+         }
+       }
+
+       // text -> JSON -> html
+       function jsonTextToHtmlList (text) {
+         var json = JSON.parse(text);
+         return Object.keys(json).map(function (key) {
+           return "<li>" + key + " : " + json[key] + "</li>"
+         }).join("");
+       }
+
+       function displayError(text) {
+         $("#error_list").html(jsonTextToHtmlList(text));
```

Comment:
> Would you consider multi-file submissions if a deploy script is also included?

Yes, I suppose it only makes sense at this point.

-----

> I suggest ignoring the existence of xhr.success and xhr.fail, and using xhr.done instead.

I had originally just used `$.ajax.done()`, but at the time, it was rather messy looking. You can still pass the jqXHR object via `success`, in this case an argument for `$.ajax()` rather than the `jqXHR.success()` function, which is itself deprecated.
```javascript
$.ajax({
    success: function(data, textStatus, jqXHR) {
        alert(jqXHR.statusCode);
    }
});
```
It's also more robust than jqXHR.done() in that it accepts multiple functions as an array:
```javascript
$.ajax({
    success: [checkStatus(), handleResponse()]
});
```
It should call each in turn with the same arguments: `data`, `textStatus`, and `jqXHR`.

----

But in any case, handling errors is even simpler than touching the jqXHR object itself.
```javascript
$.ajax({
    statusCode: {
      404: function() { alert('404'); }
    }
});